cmd/run.go

```diff
@@ -340,19 +343,21 @@ This enables a CRD-driven approach to automated image updates with Argo CD.
 	controllerCmd.Flags().StringVar(&cfg.RegistriesConf, "registries-conf-path", common.DefaultRegistriesConfPath, "path to registries configuration file")
 	controllerCmd.Flags().BoolVar(&disableKubernetes, "disable-kubernetes", false, "do not create and use a Kubernetes client")
 	controllerCmd.Flags().IntVar(&cfg.MaxConcurrency, "max-concurrency", 10, "maximum number of update threads to run concurrently")
+	controllerCmd.Flags().IntVar(&MaxConcurrentReconciles, "max-concurrent-reconciles", 1, "maximum number of concurrent Reconciles which can be run")
```

Comment:
How is `max-concurrent-reconciles` different from the existing `max-concurrency`? Is it possbile to merge them into one single flag?

Reply:
Shall we also support reading this flag from env var?

Reply:
Great catch :)
I have renamed `MaxConcurrency` to `MaxConcurrentApps` to be more specific. 
we use  `MaxConcurrentApps` and `MaxConcurrentReconciles` in different parts of the code.

```
**--max-concurrent-apps *number***

Process a maximum of *number* applications concurrently. To disable concurrent
application processing, specify a number of `1`.

**--max-concurrent-reconciles *number***

Process a maximum of *number* ImageUpdater custom resources concurrently. 
This controls how many ImageUpdater CRs can be reconciled simultaneously by the controller. 
To disable concurrent reconciliation processing, specify a number of 1. 
Higher values may improve throughput but could increase resource usage and API load.
```

Reply:
> Shall we also support reading this flag from env var?

We don't read `MaxConcurrentApps` from env vars. We can add both of them to env vars or none of them.
I don't have arguments what is better.